E dna of invasive carp gui

LoganJanes opened the pull request:
The regions in the animation have been changed to their proper coordinates and colours for a more accurate animation.
![image](https://github.com/user-attachments/assets/c2a3a4fd-fa5c-4501-8f2d-ba435830b1a6)

LoganJanes committed:
CarpTab on Gui

This is a placeholder tab for task 1.3. The page will have a section where the users can input the carp data that will be given during the competition to create an animated model as required by the task

LoganJanes committed:
CarpIcon

Adding Carp Icon for GUI page

LoganJanes committed:
CarpAnimation and Carptab functionality

Functionality for CarpTab to input data and CarpAnimation to animate the image with the inputted data. Still working on inputting data functionality

LoganJanes committed:
Input Function

Input Function for CarpTab

LoganJanes committed:
Removing input

Removing input option since the table is already given and we will know it beforehand so inputting the data is not required

LoganJanes committed:
Minor changes

Removing comments so I can add comments to all sections after changing them

LoganJanes committed:
Changes for troubleshooting

Changes to the code to attempt to resolve display issue

LoganJanes committed:
Animation Display

Code now displays the image and plays the animation when the button is clicked.

LoganJanes committed:
Proper Animation

Proper tiles are now filled in for each year in the animation

Zaid-Duraid committed:
Merge branch 'main' into eDNA-of-invasive-carp-gui

LoganJanes committed:
Colours added

Colours are added for each region

LoganJanes committed:
Regions 1 and 2 fix

Fixing location for regions 1 and 2 locations

LoganJanes committed:
Location Fixes

Fixing locations of each region

LoganJanes committed:
Transparent colours

Colours are now transparent

LoganJanes committed:
Button change

Change to the button to be displayed under the animation

Zaid-Duraid commented on web_frontend/src/components/CarpTab.tsx:
```diff
@@ -0,0 +1,161 @@
+    "Region 5": [false, false, false, false, false, false, false, false, false, false,],
+  },
+};
+
```
Would be nice to add functionality such that this data is extracted from a screenshot of the table provided at MATE.

LoganJanes replied:
Got it. I will work on this.

Zaid-Duraid commented on web_frontend/src/components/CarpAnimation.tsx:
```diff
@@ -0,0 +1,129 @@
+    [440, 60],
+    [400, 80],
+  ],
+};
```
This certainly works. It may be easier however to store 2^5 images showing each possible combination of invasive carp, and the names of these images can be something like "FFFFF.png" or "FFTFF.png".

Zaid-Duraid commented:
Overall, this is pretty good! From a competition standpoint, I think the following three things would be nice to do prior to merging this:
1. Have functionality where the program can access the camera to take a picture of the table (which we will be given on a laminated sheet) and thereby extract the data automatically
2. Make it so that in addition to the animation that is shown, each year's "carp map" can be viewed individually as well. This will make things easier for the judge because the animation is fast as it is right now. 
3. See my other comment about pre-storing 2^5 (32) images for each possible combination.

Especially for the first and third points, it may be easier to implement this program in Python. This will allow us to use OpenCV to access the camera and extract the data from the picture and read/edit/modify files on the local filesystem.

I think the Python approach would be easier, but the main thing is to hit those three points

LoganJanes committed:
Camera Functionality

Adding Camera Functionality to the page

LoganJanes committed:
Small Change

LoganJanes committed:
Updates for photo

Updates for the photo taking of the tables for data extraction

Zaid-Duraid committed:
Merge branch 'main' into eDNA-of-invasive-carp-gui

Zaid-Duraid committed:
Merge branch 'main' into eDNA-of-invasive-carp-gui

Zaid-Duraid committed:
fix: added tesseract.js to dependencies

LoganJanes committed:
Backend Implementation

Using Flask and CORS to create a backend that should help with processing the image instead of just tesseract on the frontend

LoganJanes committed:
Screenshot mode

Allows screenshots to be extracted as table data. Will be used for testing to improve the extraction code

LoganJanes committed:
Debugging

Proper debugging has been added so we can see the extracted text in the terminal as well as the image as it is processed by the OCR to see what changes are being made to it.

LoganJanes committed:
Improved detection.

Now detects most cells in the image. Adjusting the line kernel = np.ones((3,1), np.uint8) should improve detection and eventually detect all values as required. Warning that the current debugging process creates about 100 photos of debugged cells

LoganJanes committed:
Fixed Value Detection

Code now detects each cell value properly. This version of the code will produce 100 debugging images just to see what is being checked by the code from the image.

LoganJanes committed:
Lowered Detection PNGs

Now only 65 debugging images are created instead of 100.

LoganJanes committed:
Minor Comment change

Changing minor comment description

LoganJanes committed:
Animation and Display Changes

Changes to display extracted text on frontend as well as animation based on regions given Y for each year

LoganJanes committed:
Proper animation

Animation is slowed down and columns are turned green to indicate Y from the retrieved OCR data.

LoganJanes committed:
Update regions

Updated region coordinates to better reflect ideal positioning

LoganJanes committed:
Alignment changes

Changing alignment of regions

LoganJanes committed:
Update CarpTab.tsx

LoganJanes committed:
Update CarpTab.tsx

LoganJanes committed:
Merge branch 'main' into eDNA-of-invasive-carp-gui

LoganJanes committed:
Requirements for python backend

requirements for the python backend

LoganJanes committed:
Readme Change

LoganJanes committed:
Readme Change

LoganJanes committed:
Remove CV2 from requirements

LoganJanes committed:
Removing debug cells and minor frontend change

LoganJanes committed:
Manual data entry

Manual data entry for completing the task if OCR is not accurate

LoganJanes committed:
Start of region editing

Editing regions to become single lines rather than shapes

LoganJanes committed:
Pause and Resume button and region colours

Fixed pause button, added resume button and turned the region shapes into lines that change to the appropriate colours

LoganJanes committed:
Proper Region line coordinates

Proper region line coordinates have been added.

LoganJanes committed:
Update CarpTab.tsx

LoganJanes committed:
Update CarpTab.tsx

LoganJanes committed:
Added legend

Added a legend for each region colour

LoganJanes committed:
Removed Backend

Removed Backend now that animation works with frontend only

LoganJanes committed:
Removed Node Modules from root directory

Removed Node Modules from root directory
